### PR TITLE
also clean up collections, collections_entries and entity_refs when r…

### PR DESCRIPTION
**Pull request by galganif**
…emoving expired entities (temporarily uses a hack to find references for eache entity)

### CODE REVIEW

File: java/arcs/android/storage/database/DatabaseImpl.kt [hunk 1]
```diff
@@ -727,19 +727,51 @@ class DatabaseImpl(
         writableDatabase.transaction {
             // Find all expired entities.
             val storageKeyIds = rawQuery(
```
Comment (csilvestrini) [inline java/arcs/android/storage/database/DatabaseImpl.kt]: Maybe rename this to better indicate what it contains? storageKeyPairs, perhaps?

Comment (csilvestrini) [reply java/arcs/android/storage/database/DatabaseImpl.kt]: Even better might be to pull it apart into two separate lists: storageKeyIds and storageKeys

File: java/arcs/android/storage/database/DatabaseImpl.kt [hunk 2]
```diff
@@ -770,22 +802,22 @@ class DatabaseImpl(
...
+            // Remove from collection_entries all references to the expired entities.            
             delete(
                 TABLE_COLLECTION_ENTRIES,
-                "collection_id NOT IN ($usedFieldCollectionIdsQuery)",
-                arrayOf()
+                """
+                    collection_id IN (SELECT id FROM collections WHERE type_id >= ?)
+                    AND value_id NOT IN (SELECT id FROM entity_refs)
+                """.trimIndent(),
+                arrayOf(PrimitiveType.values().size.toString()) // only entity collections.
```
Comment (csilvestrini) [inline java/arcs/android/storage/database/DatabaseImpl.kt]: Might be nice to define a helper constant on PrimitiveType like PrimitiveType.LARGEST_PRIMITIVE_TYPE_ID or something similar. Could use it in isPrimitiveType too

**Comment by galganif** [comment]
\gcbrun

**Pull request merged**